contrib/opencensus-ext-fastapi/opencensus/ext/fastapi/fastapi_middleware.py

```diff
@@ -0,0 +1,168 @@
+                    :class:`.ProbabilitySampler`. Other options include
+                    :class:`.AlwaysOnSampler` and :class:`.AlwaysOffSampler`.
+
+    :type exporter: :class:`~opencensus.trace.base_exporter.exporter`
+    :param exporter: An exporter. Default to
+                     :class:`.PrintExporter`. The rest options are
+                     :class:`.FileExporter`, :class:`.LoggingExporter` and
+                     trace exporter extensions.
+
+    :type propagator: :class: 'object'
+    :param propagator: A propagator. Default to
+                       :class:`.TraceContextPropagator`. The rest options
+                       are :class:`.BinaryFormatPropagator`,
+                       :class:`.GoogleCloudFormatPropagator` and
+                       :class:`.TextFormatPropagator`.
+    """
+
+    def __init__(
+        self,
+        app: ASGIApp,
+        excludelist_paths=None,
+        excludelist_hostnames=None,
+        sampler=None,
+        exporter=None,
+        propagator=None,
+    ) -> None:
+        super().__init__(app)
+        self.excludelist_paths = excludelist_paths
+        self.excludelist_hostnames = excludelist_hostnames
+        self.sampler = sampler or samplers.AlwaysOnSampler()
+        self.exporter = exporter or print_exporter.PrintExporter()
+        self.propagator = propagator or trace_context_http_header_format.TraceContextPropagator()
+
+        # pylint: disable=protected-access
+        integrations.add_integration(integrations._Integrations.FASTAPI)
+
+    def _prepare_tracer(self, request: Request) -> tracer_module.Tracer:
+        span_context = self.propagator.from_headers(request.headers)
+        tracer = tracer_module.Tracer(
+            span_context=span_context,
+            sampler=self.sampler,
+            exporter=self.exporter,
+            propagator=self.propagator,
+        )
+        return tracer
+
+    def _before_request(self, span: Union[Span, BlankSpan], request: Request):
+        span.span_kind = span_module.SpanKind.SERVER
+        span.name = "[{}]{}".format(request.method, request.url)
+        span.add_attribute(HTTP_HOST, request.url.hostname)
+        span.add_attribute(HTTP_METHOD, request.method)
+        span.add_attribute(HTTP_PATH, request.url.path)
+        span.add_attribute(HTTP_URL, str(request.url))
+        span.add_attribute(HTTP_ROUTE, request.url.path)
+        execution_context.set_opencensus_attr(
+            "excludelist_hostnames", self.excludelist_hostnames
+        )
+
+    def _after_request(self, span: Union[Span, BlankSpan], response: Response):
+        span.add_attribute(HTTP_STATUS_CODE, response.status_code)
+
+    def _handle_exception(self, span: Union[Span, BlankSpan], exception: Exception):
+        span.add_attribute(ERROR_NAME, exception.__class__.__name__)
+        span.add_attribute(ERROR_MESSAGE, str(exception))
+        span.add_attribute(STACKTRACE, "\n".join(traceback.format_tb(exception.__traceback__)))
```

Comment:
I noticed that the span object will end up with status 0 (=OK) in case of an exception. I added `span.set_status(Status.from_exception(exception))` to ensure that downstream systems treat this as an error.

Reply:
The stack trace is also collected differently in the Flask extension: https://github.com/census-instrumentation/opencensus-python/blob/master/contrib/opencensus-ext-flask/opencensus/ext/flask/flask_middleware.py#L223-L229 

I believe this is the reason why I don't see the stack trace in the proper place in Azure Application Insights. The actual "stack_trace" attribute is empty.

Reply:
Agree with @calleo  that `status` should be set on exception.

Reply:
Thanks for pointing that out. I think I've found a bug in the azure exporter which I have reported here: https://github.com/census-instrumentation/opencensus-python/pull/1187

I believe it would be good to set the http status code as well (to 500), as that's what's going to happen if we end-up in `_handle_exception`.

Reply:
@calleo  Left a couple of comments on your pr. Good find.

Reply:
Thank you for your comments. I made span status 500 in case of exception.   https://github.com/census-instrumentation/opencensus-python/pull/1124/commits/64fd4dc991609da880382fedbd81446b6eff9178